chalk-rust-ir/src/lib.rs

```diff
@@ -422,6 +422,23 @@ pub struct AssociatedTyValueBound<I: Interner> {
     pub ty: Ty<I>,
 }
 
+/// Represents the bounds for an `impl Trait` type.
+///
+/// ```ignore
+/// opaque type T: A + B = HiddenTy;
+/// ```
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Fold)]
+pub struct ImplTraitDatum<I: Interner> {
+    /// The placeholder `!T` that corresponds to the opaque type `T`.
+    pub impl_trait_id: ImplTraitId<I>,
+
+    /// Trait bounds for the opaque type.
+    pub bounds: Vec<QuantifiedWhereClause<I>>,
+
+    /// The "hidden type" that the opaque type is equal to when revealed.
+    pub ty: Binders<Ty<I>>,
```

Comment:
let's call this `hidden_ty` (but, as I wrote, I think we'll want to remove it eventually)